### PR TITLE
servlets added

### PR DESCRIPTION
**Pull request by AndreyDMA**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -64,5 +85,4 @@
             </plugins>
         </pluginManagement>
     </build>
-
 </project>
```
Comment (Sofasmile) [inline pom.xml]: add a blank line at the end of the file

File: src/main/java/mateacademy/internetshop/IdGenerator.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package mateacademy.internetshop;
 
 public class IdGenerator {
-    private static Long idGenerator = 2019L;
+    private static Long idGenerator = 201909L;
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/IdGenerator.java]: 201909L??? why?

File: src/main/java/mateacademy/internetshop/Main.java [hunk 3]
```diff
@@ -1,26 +1,14 @@
...
 import mateacademy.internetshop.lib.Injector;
-import mateacademy.internetshop.model.Bucket;
-import mateacademy.internetshop.model.Item;
-import mateacademy.internetshop.model.User;
 import mateacademy.internetshop.service.BucketService;
 import mateacademy.internetshop.service.ItemService;
 import mateacademy.internetshop.service.OrderService;
 import mateacademy.internetshop.service.UserService;
 
 public class Main {
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/Main.java]: can we remove this class?

File: src/main/java/mateacademy/internetshop/controller/AddItemToBucketController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+import mateacademy.internetshop.lib.Inject;
+import mateacademy.internetshop.model.Bucket;
+import mateacademy.internetshop.model.Item;
+import mateacademy.internetshop.service.BucketService;
+import mateacademy.internetshop.service.ItemService;
+
+public class AddItemToBucketController extends HttpServlet {
+    private Bucket bucket = new Bucket();
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/controller/AddItemToBucketController.java]: move this live in the method

File: src/main/java/mateacademy/internetshop/controller/AddItemsController.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import mateacademy.internetshop.model.Item;
+import mateacademy.internetshop.service.ItemService;
+
+public class AddItemsController extends HttpServlet {
+    @Inject
+    private static ItemService itemService;
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/addItems.jsp").forward(req, resp);
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/controller/AddItemsController.java:20]: why do we use getRequestDispatcher() here?

File: src/main/java/mateacademy/internetshop/controller/CreateOrderController.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+public class CreateOrderController extends HttpServlet {
+    @Inject
+    private static OrderService orderService;
+    @Inject
+    private static BucketService bucketService;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Item> orderedItems = bucketService.getAllItems(0L);
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/controller/CreateOrderController.java]: 0L? what does it mean??? --> create constant in all classes
try to create your logic by next plan:
we know only userId
try to use only this constant in controllers
try to get all data by userId

File: src/main/java/mateacademy/internetshop/controller/DeleteItemFromBucketController.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import mateacademy.internetshop.service.BucketService;
+
+public class DeleteItemFromBucketController extends HttpServlet {
+    @Inject
+    private static BucketService bucketService;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        bucketService.deleteItem(0L, Long.valueOf(req.getParameter("item_id")));
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/controller/DeleteItemFromBucketController.java]: the same:)

File: src/main/java/mateacademy/internetshop/controller/GetAllItemsController.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import mateacademy.internetshop.model.Item;
+import mateacademy.internetshop.service.ItemService;
+
+public class GetAllItemsController extends HttpServlet {
+    @Inject
+    private static ItemService itemService;
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/controller/GetAllItemsController.java]: remove one blank line

File: src/main/java/mateacademy/internetshop/controller/GetAllItemsInBucketController.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    private static BucketService bucketService;
+    @Inject
+    private static UserService userService;
+    @Inject
+    private static ItemService itemService;
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
+        List<Item> items = bucketService.getAllItems(0L);
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/controller/GetAllItemsInBucketController.java]: :)

File: src/main/java/mateacademy/internetshop/controller/GetAllOrdersController.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    @Inject
+    private static UserService userService;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Order> orders = userService.getOrders(1L);
+        req.setAttribute("orders",orders);
+        req.getRequestDispatcher("/WEB-INF/views/allOrders.jsp").forward(req, resp);
+
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/controller/GetAllOrdersController.java]: remove one redundant line

File: src/main/java/mateacademy/internetshop/controller/GetAllUsersController.java [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+public class GetAllUsersController extends HttpServlet {
+    @Inject
+    private static UserService userService;
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
+        List<User> users = userService.getAll();
+        req.setAttribute("users",users);
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/controller/GetAllUsersController.java]:
```suggestion
        req.setAttribute("users", users);
```

File: src/main/java/mateacademy/internetshop/model/Bucket.java [hunk 12]
```diff
@@ -3,19 +3,23 @@
...
 public class Bucket {
-    private final Long id;
-    private List<Item> items = new ArrayList<>();
+    private final Long bucketId;
+    private List<Item> items;
     private User user;
 
     public Bucket() {
-        this.id = IdGenerator.getGeneratedId();
+        this.bucketId = 0L; //IdGenerator.getGeneratedId();
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/model/Bucket.java]: ?

File: src/main/java/mateacademy/internetshop/model/User.java [hunk 13]
```diff
@@ -1,25 +1,62 @@
...
     private Bucket bucket;
+    private String name;
+    private String surname;
+    private String password;
+    private String login;
 
     public User() {
-        this.id = IdGenerator.getGeneratedId();
-        this.orders = new ArrayList<>();
+        this.id = 1L; //IdGenerator.getGeneratedId();
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/model/User.java]: ?

File: src/main/java/mateacademy/internetshop/dao/UserDao.java [hunk 14]
```diff
@@ -8,7 +8,9 @@
 
     User get(Long id);
 
+    Long getId(User user);
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/UserDao.java]: do we need this method here?

File: src/main/java/mateacademy/internetshop/dao/daoimpl/UserDaoImpl.java [hunk 15]
```diff
@@ -36,7 +36,14 @@ public User update(User user) {
     }
 
     @Override
-    public void delete(Long id) {
-        Storage.users.removeIf(u -> u.getId().equals(id));
+    public Long getId(User user) {
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/daoimpl/UserDaoImpl.java]: the same

File: src/main/java/mateacademy/internetshop/service/UserService.java [hunk 16]
```diff
@@ -8,7 +10,13 @@
 
     User get(Long id);
 
+    Long getId(User user);
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/service/UserService.java]: :)

File: src/main/java/mateacademy/internetshop/service/serviceimpl/UserServiceImpl.java [hunk 17]
```diff
@@ -22,13 +26,28 @@ public User get(Long id) {
         return userDao.get(id);
     }
 
+    @Override
+    public Long getId(User user) {
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/service/serviceimpl/UserServiceImpl.java]: :)

File: src/main/java/mateacademy/internetshop/service/serviceimpl/UserServiceImpl.java [hunk 18]
```diff
@@ -22,13 +26,28 @@ public User get(Long id) {
...
     @Override
-    public void delete(Long id) {
-        userDao.delete(id);
+    public User delete(Long id) {
+        return userDao.delete(id);
+    }
+
+    @Override
+    public List<User> getAll() {
+        return Storage.users;
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/service/serviceimpl/UserServiceImpl.java]: we can't use db in service

File: src/main/webapp/WEB-INF/views/allItems.jsp [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+  Date: 17-Sep-19
+  Time: 18:24
+  To change this template use File | Settings | File Templates.
+--%>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>AllItems</title>
+
+    <style>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/allItems.jsp:14]: optional: better move style in  the css file

File: src/main/webapp/WEB-INF/web.xml [hunk 20]
```diff
@@ -0,0 +1,110 @@
...
+    <servlet-mapping>
+        <servlet-name>addItemToBucket</servlet-name>
+        <url-pattern>/addItemToBucket</url-pattern>
+    </servlet-mapping>
+
+    <listener>
+        <listener-class>mateacademy.internetshop.controller.InjectInitializer</listener-class>
+    </listener>
+
+</web-app>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/web.xml]: add a blank line

File: pom.xml (hunk 1 above)
Comment (AndreyDMA) [reply pom.xml]: done

File: src/main/java/mateacademy/internetshop/IdGenerator.java (hunk 2 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/IdGenerator.java]: '0' is too bored))
another reason - I have one IdGenerator for all ID's, and it looks not good with start values (0, 1, 2 …)

File: src/main/java/mateacademy/internetshop/Main.java (hunk 3 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/Main.java]: done

File: src/main/java/mateacademy/internetshop/controller/AddItemToBucketController.java (hunk 4 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/controller/AddItemToBucketController.java]: if I'll move it inside the method, then I will always create new bucket, when press refresh button

File: src/main/java/mateacademy/internetshop/controller/AddItemsController.java (hunk 5 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/controller/AddItemsController.java:20]: we use it to call addItem page (addItem.jsp), where I have some forms to add items

File: src/main/java/mateacademy/internetshop/controller/CreateOrderController.java (hunk 6 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/controller/CreateOrderController.java]: We discuss it with Bohdan... he say that in this homework we need to learn how to create servlets, so we can simplify our descision by using fixed ID's instead if userId, bucketId etc... to improve my turtle speed...
If it have to be fixed now - I'll fix it (in about 24h), if such descision is acceptable for now let it be...

File: src/main/java/mateacademy/internetshop/controller/GetAllItemsController.java (hunk 8 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/controller/GetAllItemsController.java]: done

File: src/main/java/mateacademy/internetshop/controller/GetAllItemsInBucketController.java (hunk 9 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/controller/GetAllItemsInBucketController.java]: done

File: src/main/java/mateacademy/internetshop/controller/GetAllOrdersController.java (hunk 10 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/controller/GetAllOrdersController.java]: done

File: src/main/java/mateacademy/internetshop/controller/GetAllUsersController.java (hunk 11 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/controller/GetAllUsersController.java]: fixed

File: src/main/java/mateacademy/internetshop/model/Bucket.java (hunk 12 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/model/Bucket.java]: temporary simplifying

File: src/main/java/mateacademy/internetshop/model/User.java (hunk 13 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/model/User.java]: same... I will replace it when I understand how to know from which exact user I making purchases...

File: src/main/java/mateacademy/internetshop/dao/UserDao.java (hunk 14 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/UserDao.java]: Look's like we don't)) I thinck there is a lot of such methods))). I wrote it then I change it, then add something, then remove something... and did it not once)). I am already looking for time to make review

removed

File: src/main/java/mateacademy/internetshop/dao/daoimpl/UserDaoImpl.java (hunk 15 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/daoimpl/UserDaoImpl.java]: removed

File: src/main/java/mateacademy/internetshop/service/UserService.java (hunk 16 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/service/UserService.java]: removed

File: src/main/java/mateacademy/internetshop/service/serviceimpl/UserServiceImpl.java (hunk 17 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/service/serviceimpl/UserServiceImpl.java]: removed

File: src/main/java/mateacademy/internetshop/service/serviceimpl/UserServiceImpl.java (hunk 18 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/service/serviceimpl/UserServiceImpl.java]: fixed

File: src/main/webapp/WEB-INF/views/allItems.jsp (hunk 19 above)
Comment (AndreyDMA) [reply src/main/webapp/WEB-INF/views/allItems.jsp:14]: I just want to make it well looking...

File: src/main/webapp/WEB-INF/web.xml (hunk 20 above)
Comment (AndreyDMA) [reply src/main/webapp/WEB-INF/web.xml]: done

File: src/main/java/mateacademy/internetshop/IdGenerator.java [hunk 21]
```diff
@@ -1,9 +1,11 @@
 package mateacademy.internetshop;
 
 public class IdGenerator {
-    private static Long idGenerator = 201909L;
+    private static Long idGenerator;
+    private static final Long START_VALUE = 201909L;
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/IdGenerator.java]: it isn't good practice to have one id generator, try to create separate for each entity

File: src/main/java/mateacademy/internetshop/controller/AddItemToBucketController.java (hunk 4 above)
Comment (Sofasmile) [reply src/main/java/mateacademy/internetshop/controller/AddItemToBucketController.java]: краще створити трохи іншу логіку) Виносити подібні речі в клас - погана ідея, адже при одночасній обробці кількох запитів від різних користувачів ми будемо отримувати не актуальні дані.
як варіант,  краще захардкодити UserId і діставати корзину за цим значенням. якщо корзини немає - то створити, якщо є - використати існуючу

File: src/main/java/mateacademy/internetshop/controller/CreateOrderController.java (hunk 6 above)
Comment (Sofasmile) [reply src/main/java/mateacademy/internetshop/controller/CreateOrderController.java]: так, я роpумію що це займе багато часу, але уже в сьогоднішній дз доведеться перебудувати логіку програми. Можеш покищо залишати так як є, але повинось будь ласка в константи '0L' в усіх контроллерах адже не зрозуміло що це за число (magic number)

File: src/main/java/mateacademy/internetshop/IdGenerator.java (hunk 21 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/IdGenerator.java]: fixed

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**